Randomized assignment

sjecmen opened the pull request.

melisabok commented:
The object that does the integration with the API doesn't have the probability limits implemented, could you add it?
For now, we can return the same value. Then we can call to the API to get the probabilities from edges. 

```
AttributeError: 'ConfigNoteInterface' object has no attribute 'probability_limits'
```

sjecmen commented:
I added a dummy value so that the default probability limit of 1.0 is always used (i.e., the randomized assignment will be the same as the optimal deterministic assignment). I could easily change this so that the probability limit is always 0.5 (and assignments are actually randomized), but I think that the 1.0 value should be fine for checking that the solver integration works.

melisabok commented:
Hi @sjecmen ,

I'm running the tests locally and I'm having an assert error in the line 400:
```
assert np.all(np.isclose(solution, solver.fractional_assignment_matrix))
```

I print both matrices and I see:

```
solution [[0.  0.7 0.3]
 [0.3 0.  0.7]
 [0.7 0.3 0. ]]
fractional_assignment_matrix [[0.3 0.4 0.3]
 [0.  0.3 0.7]
 [0.7 0.3 0. ]]
```

They seem to be different, what am I missing?

I had to upgrade ortools to version 8.1.8487 because `pywraplp.Solver.CreateSolver('GLOP')` didn't exist.

sjecmen commented:
Sorry, that's my mistake. I didn't see that that test case had multiple valid solutions and didn't run into the issue myself. I've fixed that case and checked over the other test cases for this issue more carefully. I also added a version requirement for ortools to the setup.py.

melisabok reviewed (approved):
Thanks for the fix!

It looks good to me. I'm going to merge this now and create a new PR with the integration with OpenReview API. 

It would nice if you can write an explanation of the solver in the README file.
